docs/site/migration/models/remoting-hooks.md

```diff
@@ -251,21 +291,238 @@ export class ProductController {
+| `ctx.args`            | `invocationCtx.args`                                                |
+| `ctx.result`          | the value returned by `await next()`                                |
+| `ctx.error`           | use `catch` to receive errors thrown by `await next()`              |
+| `ctx.req.accessToken` | see [Accessing the current user](#accessing-the-current-user) below |
+| `ctx.methodString`    | `invocationCtx.targetName`                                          |
 
 ## Modifying request parameters
 
-(to be done)
+It is possible to modify arguments passed to the controller method before the
+method is invoked.
+
+{% include warning.html content="
+If your interceptor is registered for more than a single method, then extra care is needed to ensure your interceptor is making valid assumption about the arguments expected by the target controller method.
+" %}
+
+Consider the following LoopBack 3 remoting hook executed for `Comment.create`
+method to add `postedFromIpAddress` field to every new comment posted:
+
+```js
+Comment.afterRemote('create', function(ctx, next) {
+  ctx.args[0].postedFromIpAddress = ctx.req.remoteAddress;
+  next();
+});
+```
+
+This remoting hook can be rewritten to a LoopBack 4 interceptor as follows:
+
+```ts
+class AddRemoteAddressInterceptor implements Provider<Interceptor> {
+  constructor(@inject(RestBindings.Http.REQUEST) private request: Request) {}
+
+  // ...
+
+  async intercept(
+    invocationCtx: InvocationContext,
+    next: () => ValueOrPromise<InvocationResult>,
+  ) {
+    invocationCtx.args[0].postedFromIpAddress = this.request.socket.remoteAddress;
+    return next();
+  }
+}
+```
 
 ## Modifying the response
 
-(to be done)
+You can transform the value returned by `await next()` before returning it from
+the interceptor, thus modifying the final response returned to the client.
+
+Consider the following example from LoopBack 3 documentation:
+
+```js
+// prevent password hashes from being sent to clients
+Customer.afterRemote('**', function(ctx, user, next) {
+  if (ctx.result) {
+    if (Array.isArray(ctx.result)) {
+      ctx.result.forEach(function(result) {
+        result.unsetAttribute('password');
+      });
+    } else {
+      ctx.result.unsetAttribute('password');
+    }
+  }
+
+  next();
+});
+```
+
+This remoting hook can be rewritten to a LoopBack 4 interceptor as follows:
+
+```ts
+async function intercept(
+  invocationCtx: InvocationContext,
+  next: () => ValueOrPromise<InvocationResult>,
+) {
+  const result = await next();
+  if (Array.isArray(result)) {
+    result.forEach(it => delete it.password);
+  } else {
+    delete result.password;
+  }
+  return result;
+}
+```
+
+## Rejecting requests
+
+In order to reject a request and return an error HTTP response, just throw an
+error from your interceptor. As explained in
+[Handling errors in controllers](../../Controllers.html#handling-errors-in-controllers),
```

Comment:
I don't know if `Controllers.html` should be `Controllers.md`?

Reply:
Good catch! I was copying full URLs from https://loopback.io and changing them to local paths. I have to review all links and replace `.html` to `.md` where appropriate.

Thank you for catching this problem! ❤️